iceoryx_posh/test/moduletests/test_capro_service.cpp

```diff
@@ -148,7 +148,11 @@ TEST_F(ServiceDescription_test, ServiceDescriptionSerializationCreatesServiceDes
                                                      static_cast<uint16_t>(testScope),
                                                      static_cast<uint16_t>(testInterfaceSource));
 
-    ServiceDescription serviceDescription1 = ServiceDescription(serialObj);
+
+    auto maybeService = ServiceDescription::deserialize(serialObj);
```

Comment:
To bypass the naming situation, you could do something like this
```cpp
ServiceDescription::deserialize(serialObj).and_then([&](const auto & sut){
    EXPECT_THAT(...);
    ...
}).or_else([&](const auto error){
    FAIL() << "Deserialization should not fail but failed with: " << error;
});
```

What do you think?

Reply:
I like it. It even lead to  more stringent const correctness code.